### PR TITLE
set up ignore and rc file for eslint

### PR DESCRIPTION
**Pull request by eanplatter**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      "after": true
+    }],
+    "space-after-keywords": 2,       // http://eslint.org/docs/rules/space-after-keywords
+    "space-before-blocks": 2,        // http://eslint.org/docs/rules/space-before-blocks
+    "space-before-function-paren": [2, "never"], // http://eslint.org/docs/rules/space-before-function-paren
+    "space-infix-ops": 2,            // http://eslint.org/docs/rules/space-infix-ops
+    "space-return-throw-case": 2,    // http://eslint.org/docs/rules/space-return-throw-case
+    "spaced-line-comment": 2         // http://eslint.org/docs/rules/spaced-line-comment
+  }
+}
```
Comment (eanplatter) [inline .eslintrc]: taken from the https://github.com/technologyadvice/vers project.

File: test/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "env": {
+    "node": true,
+    "mocha": true
+  },
+  "rules": {
+    "no-console": 0,                // http://eslint.org/docs/rules/no-console
+    "new-cap": 0                    // http://eslint.org/docs/rules/new-cap
+  }
+}
```
Comment (eanplatter) [inline test/.eslintrc]: a special configuration for the test folder allows us to reduce trivial errors

**Comment by levithomason** [comment]
From the ESlint integrations page: https://www.npmjs.com/package/gulp-eslint

Yea yea!

**Comment by eanplatter** [comment]
༼ つ ◕_◕ ༽つ fine I'll do my own emoji

**Pull request merged**